runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -234,8 +227,9 @@ public void write(IToken token, Object data) throws OverwriteException {
         if (!runtime.getParameters().isCacheDisabled()) {
             ILogData data;
             try {
-                data = readCache.get(address);
-            } catch (ExecutionException | UncheckedExecutionException e) {
+                data = readCache.asMap().computeIfAbsent(address, this::fetch);
```

Comment:
Maybe a comment here why using computeIfAbsent and how it helps with the race condition.